src/safeposix/syscalls/net_calls.rs

```diff
@@ -639,13 +664,15 @@ impl Cage {
                 let objectid = &sockfdobj.socketobjectid;
 
                 if let Some(localaddr) = sockfdobj.localaddr.as_ref().clone() {
-                    let release_ret_val = mutmetadata._release_localport(localaddr.addr(), localaddr.port(), sockfdobj.protocol, sockfdobj.domain);
+                    let release_ret_val = mutmetadata._release_localport(localaddr.addr(), localaddr.port(), sockfdobj.protocol, sockfdobj.domain, fd);
                     if let Err(e) = release_ret_val {return e;}
                 }
                 if !partial {
                     if let None = objectid {} else {
                         mutmetadata.socket_object_table.remove(&objectid.unwrap());
                     }
+                    //hmm but what if there are multiple fds referring to the same socket that need
```

Comment:
You left yourself a note here. Did this get clarified?

Reply:
Yeah I ended up fixing this a while ago.